### PR TITLE
feat: upgrade to Polkadot v0.9.23

### PR DESCRIPTION
**Pull request by wischli**
## fixes KILTProtocol/ticket#2023
* Upgrades to Polkadot v0.9.23
* Removes deprecated migrations
* Adds check for non-zero AccountID to all runtimes (https://github.com/paritytech/substrate/pull/10413)
* Upgrades to latest nightly [supported by Polkadot](https://github.com/paritytech/polkadot/releases/tag/v0.9.23): 2022-05-11
* Bumps srtool to 1.60
* Fixes RILT Spec
* Improves error message for incorrect parachain runtime/spec commands

## TODO

- [x] Check Rust version (latest nightly 2022-05-31 for now)
- [x] Check clients
- [x] Make sure our custom RPC was not removed (big RPC refactor)
- [x] Tripple check migrations (should be none required for us)
- [x] Run benchmarks for `pallet_treasury` and `pallet_utility` (new functions added to API)
- [x] Check whether CI is upgraded to newer nightly as well

## Checklist:

- [x] I have verified that the code works
  - [x] No panics! (checked arithmetic ops, no indexing `array[3]` use `get(3)`, ...)
- [x] I have verified that the code is easy to understand
  - [ ] If not, I have left a well-balanced amount of inline comments
- [x] I have [left the code in a better state](https://deviq.com/principles/boy-scout-rule)
- [x] I have documented the changes (where applicable)

### CODE REVIEW

**Review by ntn-x2 — APPROVED** [review]
LGTM! I see that they have also switched to using [srtool 1.60](https://github.com/paritytech/srtool/releases/tag/v0.9.20), which we might want to upgrade as well. Not sure if within the scope of this PR, but if not please make a ticket for it.

File: nodes/parachain/src/rpc.rs [hunk 1]
```diff
@@ -57,29 +59,24 @@ where
...
 
-	let mut io = jsonrpc_core::IoHandler::default();
+	let mut module = RpcExtension::new(());
 	let FullDeps {
 		client,
 		pool,
 		deny_unsafe,
 	} = deps;
 
-	io.extend_with(SystemApi::to_delegate(FullSystem::new(
```
Comment (wischli) [inline nodes/parachain/src/rpc.rs:70]: The `FullSystem` is the important part here. We use(d) this for the `NextNonce` call. It was refactored to `SystemRpc` in [Substrate #8783](https://github.com/paritytech/substrate/pull/8783/files#diff-079bdf5b7d7db5208076eef2e7d1aec9ff00ccbb3c947ddda949ee9ceb0a6347R83) and is now named [`System`](https://github.com/paritytech/substrate/blob/polkadot-v0.9.23/utils/frame/rpc/system/src/lib.rs#L81).

**Review by trusch — APPROVED** [review]
LGTM!

**Pull request merged**